proto/tendermint/types/types.proto

```diff
@@ -223,9 +223,18 @@ message TxProof {
 // transaction. This allows for tendermint to track malleated and original
 // transactions
 message MalleatedTx {
```

Comment:
Yeah, sounds good!!

Reply:
After thinking about this more, I think `BlockTx`, `BlockBlobTx`, or `ShareIndexedTx` may be more precise than `WrappedTx`. The rationale being that both `BlobTx` and the transaction we're evaluating names for (i.e. `MalleatedTx`) are wrapped and unwrapped at different stages in the transaction flow because both types of transactions contain an inner transaction (henceforth referred to as `MsgPayForBlobTx`). 

`BlockTx` or `BlockBlobTx` attempt to convey that this transaction type gets included on-chain in a block.

`ShareIndexedTx` only makes sense if we don't expect to add other metadata to this tx.

Ref: https://github.com/celestiaorg/celestia-app/pull/1005/files#diff-e1480f76524405adf851f7b9d0ca187c9e8b24e3211801bd4d02636b51aa5141R74

Reply:
`ShareIndexedTx` works for me

Reply:
[not blocking] The rationale for using `BlockTx` or `ShareIndexedTx` instead of `WrappedTx` is that the transaction that gets submitted to the mempool is also "wrapped" but with Blobs and not share indexes.

Since we're talking about three transactions, my latest thinking on naming:
1. BlobTx - is an sdk.Tx with an associated Blob. It is submitted by user to mempool
2. sdk.Tx - is a component of both BlobTx and BlockTx
3. BlockTx - is an sdk.Tx with an associated share index. It gets included in a block

Reply:
I see the logic in BlockTx, but personally find it a bit confusing as everything in the BlobTx also makes it in the block, and we're using the hash of the sdk.Tx as a reference for everything. 

More importantly tho Blo...Tx and Blo..Tx lol I never read the middle of words!! The only difference to my eye is one is slightly shorter which makes it difficult to distinguish :laughing:

Reply:
`WrappedTx` is a struct already used in the prioritized mempool jus FYI.

Reply:
Just for my understanding, when a `BlobTx` is staged for inclusion in a block it gets modified into this `MalleatedTx` (which really just wraps around it) because we also need to include which share index it begins in the block?

Reply:
What other metadata do we imagine being included here?

Reply:
> Just for my understanding, when a BlobTx is staged for inclusion in a block it gets modified into this MalleatedTx (which really just wraps around it)

That's the idea but I want to point a small detail. The BlobTx gets split into it's component parts (Blob and sdk.Tx) during block construction. A BlockTx wraps around the sdk.Tx and adds a share index. Notably: it does not wrap around the BlobTx because the BlobTx contains the entire Blob. We split the Blob and sdk.Tx so that we can write the sdk.Tx to transaction shares and the Blob to sparse shares. This enables parties who are interested in the state transitions on Celestia (IIUC most types of nodes) to download and process  all transaction shares without forcing them to download all blobs.

Ref: https://github.com/celestiaorg/celestia-app/pull/1005/files#diff-e1480f76524405adf851f7b9d0ca187c9e8b24e3211801bd4d02636b51aa5141R71

Reply:
> What other metadata do we imagine being included here?

eventually we will add either the intermediate state root itself or the index to the intermediate state root similar to the original [specs](https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/data_structures.md#wrappedtransaction)

cc @adlerjohn on a related note, can we assume that the same order between each transaction and its intermediate state root?

Reply:
Ok, last question: Is this struct intended to be used for all original transactions that are submitted to a node or just for `BlobTx`?

Reply:
> can we assume that the same order between each transaction and its intermediate state root?

Yes, that was the intention.